core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -2076,29 +2076,34 @@ class KafkaApis(val requestChannel: RequestChannel,
       // When this callback is triggered, the remote API call has completed
       request.apiRemoteCompleteTimeNanos = time.nanoseconds
     }
-    val quotaSensors = quotas.request.getOrCreateQuotaSensors(request.session.sanitizedUser, clientId)
-    def recordNetworkThreadTimeNanos(timeNanos: Long) {
-      quotas.request.recordNoThrottle(quotaSensors, nanosToPercentage(timeNanos))
-    }
-    request.recordNetworkThreadTimeCallback = Some(recordNetworkThreadTimeNanos)
+    if (quotas.request.quotasEnabled) {
```

Comment:
Could we move the conditional logic to `quotas`? The fact we ask `quotas.request` if it's enabled and then call `quotas.request.getOrCreateQuotaSensors` seems to imply that we should be able to simplify this some. One option is something like:

```scala
val blah = quotas.request.foo
request.recordNetworkThreadTimeCallback = blah.callback
blah.maybeRecordAndThrottleOnQuotaViolation(....)
```

Another is a method like the following that also takes a function for installing `recordNetworkThreadTimeCallback`.

```scala
quotas.request.maybeRecordAndThrottleQuotaViolation(...)
```

This may be the simplest option although I didn't work out all the details.